src/Indigo/Contracts/HomebaseLite/Impl/Admin.hs

```diff
@@ -15,53 +16,86 @@ import Indigo hiding ((*))
 
 import Morley.Util.Interpolate (itu)
 
-setAdmin :: tp :~> Address => IndigoEntrypoint tp
-setAdmin _ = do
+import Indigo.Contracts.HomebaseLite.Impl.Utils
+import Indigo.Contracts.HomebaseLite.Types
+
+checkSenderIsAdmin :: HasStorage Storage => IndigoProcedure
+checkSenderIsAdmin = defFunction do
+  when (sender /= storage #! #sAdmin) do
+    failCustom_ @() #senderIsNotAdmin
```

Comment:
Is there a particular reason to fail with a pair unit? AFAIU, there is `failCustomNoArg` which will fail with the errror string directly.

Reply:
I'm just following [indigo tutorial](https://indigo-lang.gitlab.io/side-effects/) in this case.

Also, why do we even have `failCustom_` to begin with?

Reply:
The `failCustom_` function comes first and `failCustomNoArg` is a recent addition. I guess the tutotrial is not updated with it.

The original issue is this https://gitlab.com/morley-framework/morley/-/issues/383, I think we should use `failCustomNoArg` since it saves some contract size.

Reply:
So, I did that in 14224cdc4acb60b97a96c35462ee343aa9d5ad42, but there are a few snags:

- `#senderIsNotAdmin` is defined in lorentz and exported from indigo, and it is defined as a pair. Which means we can't use this name with `failCustomNoArg`.
- `errorDoc` doesn't play with `failCustomNoArg`, because it explicitly specifies `type instance ErrorArg` as `()`. And writing out all the instances by hand is rather tedious, so now I'm duplicating code from `Lorentz.Util.TH` :unamused: 

I'll go raise these as issues on morley.